mono/unit-tests/test-mono-string.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * test-mono-string.c: Unit test for runtime MonoString* manipulation.
+ */
+
+#include "config.h"
+#include <stdio.h>
+#include "metadata/object-internals.h"
+#include "mini/jit.h"
+
+static int
+new_string_ok (void)
+{
+	MonoError error;
+	MonoString *s = mono_string_new_checked (mono_domain_get (), "abcd", &error);
+	mono_error_assert_ok (&error);
+	gunichar2* c = mono_string_chars (s);
+
+	g_assert ((c[0] == (gunichar2)'a') &&
```

Comment:
No need for the casts. This doesn't work on host=ebcdic systems which we probably don't care about. I violate that elsewhere also.